src/main/java/uk/gov/hmcts/reform/lrdapi/service/impl/RegionServiceImpl.java

```diff
@@ -18,13 +36,103 @@
+        if (regionId.strip().equalsIgnoreCase(ALL)) {
+            return retrieveAllRegions(false);
         }
 
-        return new LrdRegionResponse(region);
+        //otherwise generate list of IDs from regionId String
+        List<String> regionsIdList =
+            checkIfValidCsvIdentifiersAndReturnList(regionId, EXCEPTION_MSG_NO_VALID_REGION_ID_PASSED);
+
+        //then check if List contains 'ALL' and the ID for National
+        if (isListContainsTextIgnoreCase(regionsIdList, ALL) && regionsIdList.contains("1")) {
+            return retrieveAllRegions(true);
+        } else if (isListContainsTextIgnoreCase(regionsIdList, ALL)) {
+            return retrieveAllRegions(false);
+        }
+
+        //remove invalid IDs from the list
+        checkForInvalidIdentifiersAndRemoveFromIdList(
+            regionsIdList, ALPHA_NUMERIC_REGEX, log, loggingComponentName, EXCEPTION_MSG_NO_VALID_REGION_ID_PASSED);
+
+        List<Region> regions = regionRepository.findByRegionIdIn(regionsIdList);
+
+        if (isEmpty(regions)) {
+            throw new ResourceNotFoundException("No Region(s) found with the given Region ID: " + regionsIdList);
+        }
+
+        return generateResponseList(regions);
+    }
+
+    public List<LrdRegionResponse> retrieveRegionByRegionDescription(String description) {
+        checkStringContainsMoreThanOneConsecutiveComma(description);
+
+        //generate list of Descriptions from description String
+        List<String> regionDescriptionsList =
+            checkIfValidCsvIdentifiersAndReturnList(description, EXCEPTION_MSG_NO_VALID_REGION_DESCRIPTION_PASSED);
+
+        //remove invalid Regions from the list
+        checkForInvalidIdentifiersAndRemoveFromIdList(
+            regionDescriptionsList, REGION_NAME_REGEX, log,
+            loggingComponentName, EXCEPTION_MSG_NO_VALID_REGION_DESCRIPTION_PASSED
+        );
+
+        List<Region> regions = regionRepository.findByDescriptionInIgnoreCase(regionDescriptionsList);
+
+        if (isEmpty(regions)) {
+            throw new ResourceNotFoundException("No Region(s) found with the given Region Description(s): "
+                                                    + regionDescriptionsList);
+        }
+
+        return generateResponseList(regions);
+    }
+
+    public List<LrdRegionResponse> retrieveAllRegions(boolean isNationalRequired) {
+        List<Region> regions;
+
+        if (isNationalRequired) {
+            regions = regionRepository.findAll();
+        } else {
+            regions = regionRepository.findByDescriptionNotIgnoreCase("National");
+        }
+
+        return generateResponseList(regions);
+    }
+
+    private void isRegionIdParamValid(String param) {
+        checkStringContainsMoreThanOneConsecutiveComma(param);
+
+        if (param.toUpperCase().contains(ALL) && doesStringContainAlphabetOtherThanAll(param)) {
+            throw new InvalidRequestException("The only non-numeric value allowed is 'ALL' (case insensitive)");
```

Comment:
message can be put in constant

Reply:
Addressed here: https://github.com/hmcts/rd-location-ref-api/pull/273/commits/76afb26e1b85bd0d7761d13a605b3be4bdd02680